### PR TITLE
init repo

### PR DESCRIPTION
**Pull request by hawkingrei**
add TiDB dialect for Django

### CODE REVIEW

File: django_tidb/base.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        return {
+            'version': row[0],
+            'sql_mode': row[1],
+            'default_storage_engine': row[2],
+            'sql_auto_is_null': bool(row[3]),
+            'lower_case_table_names': bool(row[4]),
+            'has_zoneinfo_database': bool(row[5]),
+        }
+
+    @cached_property
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply django_tidb/base.py:83]: this just gets tidb version and then control feature switch according to rule. so if tidb-server is upgraded, some features will be enabled

File: django_test_suite.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
Comment (tisonkun) [inline django_test_suite.sh:1]: We need a license header for script also

File: django_tidb/features.py [hunk 3]
```diff
@@ -0,0 +1,438 @@
+#
```
Comment (tisonkun) [inline django_tidb/features.py:1]: We don't need a blank line. Ditto others.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import io
```
Comment (tisonkun) [inline setup.py:1]: we need a license header here also

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[flake8]
```
Comment (tisonkun) [inline setup.cfg:1]: see also https://github.com/apache/spark/blob/master/conf/fairscheduler.xml.template, if it is possible, we need a license header here also

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,24 @@
+FROM ubuntu:latest
```
Comment (tisonkun) [inline Dockerfile:1]: we need a license header here also

File: django_test_suite.sh (hunk 2 above)
Comment (tisonkun) [reply django_test_suite.sh:1]: see also https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/bin/docker-image-tool.sh

File: tidb_settings.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+DATABASES = {
```
Comment (tisonkun) [inline tidb_settings.py:1]: we need a license header here also

**Review by tisonkun — COMMENTED** [review]
We need a license file as https://github.com/pingcap/tidb/blob/master/LICENSE

**Comment by tisonkun** [comment]
by the way, this PR is far from "chore"

File: setup.py (hunk 4 above)
Comment (hawkingrei) [reply setup.py:14]: Done

File: setup.cfg (hunk 5 above)
Comment (hawkingrei) [reply setup.cfg:14]: Done

File: Dockerfile (hunk 6 above)
Comment (hawkingrei) [reply Dockerfile:14]: Done

File: tidb_settings.py (hunk 7 above)
Comment (hawkingrei) [reply tidb_settings.py:14]: Done

File: django_tidb/features.py (hunk 3 above)
Comment (hawkingrei) [reply django_tidb/features.py]: Done

File: django_test_suite.sh (hunk 2 above)
Comment (hawkingrei) [reply django_test_suite.sh:1]: Done

**Comment by hawkingrei** [comment]
> by the way, this PR is far from "chore"

Have removed it.

> We need a license file as https://github.com/pingcap/tidb/blob/master/LICENSE

Added it

File: version.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#
```
Comment (tisonkun) [inline version.py:1]:
```suggestion
```

Comment (hawkingrei) [reply version.py]: DONE

File: run_testing_worker.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC.
```
Comment (tisonkun) [inline run_testing_worker.py:3]: I think we already make changes in L23-24?

```python
#print("Starting tidb-server with logging to /tidb.log")
#subprocess.Popen(["/tidb-server", "-log-file", "/tidb.log"])
```

Comment (tisonkun) [reply run_testing_worker.py:3]: You can take a look at https://github.com/tikv/client-go/blob/d535b62b62f97dc6513e0c41fefe009c6120807d/kv/key.go#L1-L31

Comment (hawkingrei) [reply run_testing_worker.py:3]: Done

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by sykp241095** [comment]
LGTM

**Review by bb7133 — APPROVED** [review]

**Pull request merged**